db/system_keyspace.cc

```diff
@@ -765,9 +792,6 @@ static constexpr auto schema_gc_grace = std::chrono::duration_cast<std::chrono::
         // comment
         "Scylla specific information about the local node"
        );
-       // scylla_local may store a replicated tombstone related to schema
-       // (see `make_group0_schema_version_mutation`), so we use nonzero gc grace.
-       builder.set_gc_grace_seconds(schema_gc_grace);
```

Comment:
This can be done only if raft/group0 is enabled. Same for tombstone gc mode "repair". Is raft enabled unconditionally now @kbr-scylla, i.e. there is no way to disable it?

Reply:
It is through RECOVERY mode, but that should be considered temporary. So tombstones will perhaps not be cleared while we are doing recovery but I wouldn't consider it a problem.

A bigger problem is that it is theoretically possible to stay in gossip mode and never upgrade the cluster to raft-based topology. Then tombstones will accumulate indefinitely. **However**, our documentation currently says that upgrade-to-raft-topology is a mandatory step, by explicit request from @tzach, if I remember correctly. But recently @avikivity gave a different opinion. All in all, I have no idea what the product team wants at this point.

If we consider upgrade-to-raft-topology mandatory, and someone doesn't do it, and they start accumulating tombstones, well; that will force them to do the upgrade after all, which will allow the tombstones to get cleared.

Reply:
At this stage Raft is mandatory.

@kbr-scylla if it still requires a mandatory step, then we should at least have new nodes reject joining to clusters without Raft, so the upgrade process will have to happen on the older code, and newer code won't have to test raftlessness.

Reply:
Raft (introduced in 5.2) is not enough, this feature requires **topology on Raft** (introduced in 6.0 and requiring a manually triggered upgrade procedure).

Reply:
> Raft (introduced in 5.2) is not enough, this feature requires topology on Raft (introduced in 6.0 and requiring a manually triggered upgrade procedure).


Actually the current implementation does not seem to require topology on raft anymore. (The original design, based on topology coordinator, did)